hedera-node/docs/design/services/smart-contract-service/hedera-account-service-isAuthorized-design.md

```diff
@@ -0,0 +1,136 @@
+# HIP-632 (Hedera Account Service) `isAuthorized{Raw}` design
+
+This note addresses the implementation of `isAuthorized` and `isAuthorizedRaw` of 
+[HIP-632](https://hips.hedera.com/hip/hip-632) - "Hedera Account Service (HAS) 
+System Contract".
+
+## Preliminary notice w.r.t. "virtual addresses"
+
+HIP-632 as written refers a lot to 
+[HIP-631](https://hips.hedera.com/hip/hip-631) - "Account Virtual Addresses".
+However, that HIP, though `Accepted` will not be implemented.  (Or at least, not
+at any time soon.)  So HIP-632 needs to be read with that in mind.  Specifically, 
+anywhere that there's a provision for dealing with "virtual addresses", of which there
+may be several/many per Hedera account, there is in fact only _one_: the _alias_.
+
+* This mainly affects the method `getVirtualAddresses` which will return an array, but
+it will have _at most one_ EVM address in it.
+* An issue is present to edit HIP-632 to clarify that it is dealing with aliases, not virtual addresses.  
+  
+## Phased implementation
+
+The first release of HIP-632 will include only `isAuthorizedRaw`.
+
+## Where does this go?
+
+A "hedera account service" system contract is being set up in
+`com.hedera.node.app.service.contract.impl.exec.systemcontracts.has`, and also at
+`...contract.impl.exec.processors.HasTranslatorsModule` (for the individual methods).
+
+Following the existing pattern for HAS each method will be given a package under 
+`...contract.impl.exec.systemcontracts.has` with two classes:
+* `[method]Translator` - which first confirms that the call is for the given method, and
```

Comment:
Technically we are not actually creating a TransactionBody in this case.  We normally do not if the function is a query type that does not change state and thus does not need to dispatch a TransactionBody to another service.

Reply:
I'll clarify that.